lgc/patch/MeshTaskShader.cpp

```diff
@@ -168,7 +168,8 @@ void MeshTaskShader::processMeshShader(Function *entryPoint) {
 // @param readTy : Type of value to read
 // @param byteOffset : Byte offset within the payload entry
 Value *MeshTaskShader::readTaskPayload(Function *entryPoint, Type *readTy, Value *byteOffset) {
-  auto payloadRingBufDesc = m_pipelineSysValues.get(entryPoint)->getTaskPayloadRingBufDesc();
+  auto payloadRingBufDesc =
+      m_pipelineSysValues.get(entryPoint, getShaderStage(entryPoint) == ShaderStageMesh)->getTaskPayloadRingBufDesc();
```

Comment:
Could we move this condition inside ShaderSystemValues? Then we don't need the interface change on it.

Reply:
Ah. This is a good question. My initial change was like what you expected:  check mesh stage and keep the interface unchanged. But later, I realized we might need this in NggPrimShader. In that code place, you can not tell whether a stage marked as "Geometry" is for a separate stage or is actually a merged shader. So I add a new explicit parameter. Of course, I could get rid of this consideration and follow your suggestion at present. We can further discuss the interface when there is an example after NggPrimShader rework.